core/src/providers/llm.rs

```diff
@@ -293,12 +295,26 @@ impl LLMRequest {
                 );
                 Ok(c)
             }
-            Err(e) => Err(anyhow!(
-                "Error querying `{}:{}`: error={}",
-                self.provider_id.to_string(),
-                self.model_id,
-                e.to_string(),
-            )),
+            Err(e) => {
+                let string_error = e.to_string();
+                if let Ok(e) = e.downcast::<ModelError>() {
+                    //  Log the request_id if it exists.
+                    if let Some(request_id) = e.request_id {
+                        error!(
+                            provider_id = self.provider_id.to_string(),
+                            model_id = self.model_id,
+                            request_id = request_id,
+                            "Too many retries: error=ModelError"
```

Comment:
Why Too many retries?

Reply:
We end up here if `with_retryable_back_off` failed right ?

Reply:
Happy to change it back to "Failed to query" though

Reply:
Generally yes that string is already injected in the error message. (see provider.rs)

Which means you'll likely won't have a ModelError here ever also.

Which means you want that log in `with_retryable_back_off` IMHO ?

Reply:
Indeed 👍